### PR TITLE
Added StoreMap and GetMap function

### PR DESCRIPTION
**Pull request by danielpygo**
- Added a test for these two functions
- Renamed DeleteTest3 to a more descriptive name

In this Pull Request I am adding two methods, and one test for them

### CODE REVIEW

File: etcd_helper/etcdfuncs.go [hunk 1]
```diff
@@ -74,6 +74,27 @@ func (o Etcdreader) GetList(resourceKey string) ([]string, error) {
...
 }
 
+// GetMap ...
+// Method to Get a map for a key
+func (o Etcdreader) GetMap(resourceKey string) (map[string]interface{}, error) {
+	c, err := getClient(o.EtcdServiceURL)
+	kapi := client.NewKeysAPI(c)
+
+	valuesMap := make(map[string]interface{}, 0)
+	var currentListString string
```
Comment (devdattakulkarni) [inline etcd_helper/etcdfuncs.go]: currentListString -> currentMapString

File: etcd_helper/etcdfuncs.go [hunk 2]
```diff
@@ -74,6 +74,27 @@ func (o Etcdreader) GetList(resourceKey string) ([]string, error) {
...
+
+	resp, err1 := kapi.Get(context.Background(), resourceKey, nil)
+	if err1 != nil {
+		return nil, err1
+	} else {
+		currentListString = resp.Node.Value
```
Comment (devdattakulkarni) [inline etcd_helper/etcdfuncs.go]: currentListString -> currentMapString

File: etcd_helper/etcdfuncs.go [hunk 3]
```diff
@@ -118,6 +139,30 @@ func (o Etcdwriter) StoreList(resourceKey string, dataList []string) error {
 	return nil
 }
 
+// StoreMap ...
+// Method to Store a Map into a key in json format
```
Comment (devdattakulkarni) [inline etcd_helper/etcdfuncs.go]: Method to Store a Map for a key in json format

File: etcd_helper/tests/etcd_test.go [hunk 4]
```diff
@@ -95,3 +95,28 @@ func TestDelete3(t *testing.T) {
 	assert.Equal(t, err.Error()[0:18], "100: Key not found", "The key should be deleted")
 	assert.Equal(t, "", str, "A Delete of a key/val pair stored as a list with StoreList")
 }
+
+func TestStoreMap(t *testing.T) {
+	dataList := make(map[string]interface{}, 0)
```
Comment (devdattakulkarni) [inline etcd_helper/tests/etcd_test.go]: dataList -> dataMap

**Comment by devdattakulkarni** [comment]
Some minor suggestions for variable name changes, other than that PR looks fine.

**Comment by devdattakulkarni** [comment]
In the commit message, change 'methos' to 'method'. You can delete the line that talks about operator-manager and operator-deployer.

**Comment by danielpygo** [comment]
Ok it is ready

**Pull request merged**